ninja-core/src/main/java/ninja/bodyparser/BodyParserEnginePost.java

```diff
@@ -62,21 +75,34 @@
                     field.setAccessible(true);
 
                     String[] values = ent.getValue();
-                    Object convertedValue;
-
-                    // convert the values based on the field type,
-                    // supported types are collections, arrays and boxed primities
-
-                    if (Collection.class.isAssignableFrom(fieldType)) {
-                        convertedValue = SwissKnife.convertCollection(values, getGenericType(field));
+                    
+                    if (Collection.class.isAssignableFrom(fieldType) || List.class.isAssignableFrom(fieldType)) {
+                    	
+                    	ListParamParser<?> parser = (ListParamParser<?>) ParamParsers.getListParser(getGenericType(field));
```

Comment:
I guess getListParser should get the injector... (Is that something unit tests should have caught?)